<launch>
    <arg name="sim_time_required" default="false"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>
    <arg name="debug" default="false"/>
    <arg name="use_lidar_pointcloud" default="false"/>
    <arg name="zmq_ip" default="127.0.0.1"/>
    <arg name="robot_name" default="enyo"/>
    <arg name="use_gt_frame" default="true"/>

    <arg name="robot_frame" default="base_link"/>
    <arg name="odom_frame" default="dragonfly67/odom"/>
    <arg name="map_frame" default="dragonfly67/map"/>
    <arg name="sensor_frame" default="camera_color_frame"/>
    <arg name="use_compressed_transport" default="false"/>
    <arg name="run_data_source" default="true"/>
    <arg name="use_image_interface" default="false"/>
    <arg name="load_static_tfs" default="false"/>

    <!--arg name="labelspace_name" default="ade20k_mit"/-->
    <arg name="labelspace_name" default="ade20k_indoor"/>
    <arg name="semantic_map_dir" default="$(find semantic_inference)/config/colors/"/>
    <!--arg name="semantic_map_file" default="ade20k_mit.csv"/-->
    <arg name="semantic_map_file" default="ade20k_indoor.csv"/>
    <arg name="semantic_map_path" default="$(arg semantic_map_dir)/$(arg semantic_map_file)"/>

    <arg name="pointcloud_topic" default="semantic_pointcloud"/>
    <arg name="rgb_info_topic" default="/camera/aligned_depth_to_color/camera_info"/>
    <arg name="rgb_topic" default="/camera/color/image_raw"/>
    <arg name="depth_info_topic" default="/camera/depth/camera_info"/>
    <arg name="depth_topic" default="/camera/depth/image_rect_raw"/>
    <arg name="label_topic" default="semantic/image_raw"/>

    <arg name="reconstruction_config_path" default="$(find hydra)/config/phoenix/pennquad_reconstruction_config.yaml"/>
    <arg name="frontend_config_path" default="$(find hydra)/config/phoenix/pennquad_frontend_config.yaml"/>
    <arg name="backend_config_path" default="$(find hydra)/config/phoenix/pennquad_backend_config.yaml"/>

    <include file="$(find hydra_ros)/launch/static_tfs/pennquad_static_tfs.xml" />

    <include file="$(find semantic_inference_ros)/launch/semantic_inference.launch" pass_all_args="true">
        <arg name="enable_rectification" default="true"/>
         <arg name="verbose" value="false"/>
         <arg name="rgb_image_transport" value="raw"/>
         <arg name="enable_pointcloud" value="$(eval not arg('use_image_interface'))"/>
         <arg name="pointcloud_queue_size" value="40"/>
         <arg name="model_name" value="mobilenetv2_360_640"/>
    </include>

    <include file="$(find hydra_ros)/launch/hydra.launch" pass_all_args="true">
        <arg name="dataset_name" value="penn_quad"/>
        <arg name="zmq_ip" value="$(arg zmq_ip)"/>
        <arg name="config_dir" default="$(find hydra)/config/phoenix"/>
        <arg name="kimera_sensor_filepath"
             value="$(find hydra_vio_configs)/config/robots/penn/LeftCameraParams.yaml"/>
        <arg name="use_oriented_bounding_boxes" default="true"/>
        <arg name="rgb_image_transport" value="raw"/>
        <arg name="depth_topic" default="/camera/depth/image_rect_raw"/>
        <arg name="use_outdoor_places" value="false"/>
    </include>

    <!-- visualizer for converting dsg to visualization markers -->
    <arg name="start_visualizer" default="true" doc="start hydra visualizer"/>
    <group if="$(arg start_visualizer)">
        <remap from="hydra_dsg_visualizer/dsg" to="hydra_ros_node/backend/dsg"/>
        <include file="$(find hydra_visualizer)/launch/hydra_streaming_visualizer.launch" pass_all_args="true">
            <arg name="visualizer_frame" default="$(arg map_frame)"/>
        </include>
    </group>

    <!-- rviz -->
    <arg name="start_rviz" default="true" doc="start rviz"/>
    <group if="$(arg start_rviz)">
        <include file="$(find hydra_visualizer)/launch/hydra_rviz.launch">
            <arg name="rviz_dir" default="$(find hydra_ros)/rviz"/>
            <arg name="rviz_file" default="penn_quad.rviz"/>
        </include>
    </group>
</launch>
